### ISSUE TITLE
Cookie name validation is overly strict

### ISSUE THREAD

**Issue by p-himik** [cmbuckley/setcookie.net#11]
The current `pattern` is set to `[A-Za-z0-9_-]+`. It doesn't allow e.g. dots, which are definitely allowed in cookie names.

**Comment by cmbuckley-nhs** [comment]
Hi - as you'll have probably seen on the site itself, this is purposefully strict compared to the spec, mainly for simplicity's sake at the time. It's in the `pattern` attributes and also the server-side validation: https://github.com/cmbuckley/setcookie.net/blob/b3ede372b62e1bd8a66cf287399f9ff242392599/public/index.php#L63

I'd welcome a pull request if you want to relax the restriction a bit!

**Comment by cmbuckley** [comment]
Hi - as you'll have probably seen on the site itself, this is purposefully strict compared to the spec, mainly for simplicity's sake at the time. It's in the `pattern` attributes and also the server-side validation: https://github.com/cmbuckley/setcookie.net/blob/b3ede372b62e1bd8a66cf287399f9ff242392599/public/index.php#L63

I'd welcome a pull request if you want to relax the restriction a bit!

**Comment by p-himik** [comment]
Oh wow, PHP makes it _much_ harder than it should be.

I started off with actual regexes for cookie name and value fields (while still not allowing stuff like newlines and vertical tabs):
```php
define("SAFE_NAME_RE_SET", "A-Za-z0-9!#$%&'*+\\-.\\^_`\\|~");
define("SAFE_VALUE_RE_SET", "A-Za-z0-9!#$%&'()*+\\-.\\/:<=>?@\\[\\\\\\]\\^_`{\\|}~");
```

But then I learned that keys in `$_COOKIE` have `.` automatically converted to `_` because apparently there's not enough suffering in the world. Oh well, here's to not having to deal with PHP for another decade or so.
